src/main/java/com/arpnetworking/metrics/common/sources/TcpLineSource.java

```diff
@@ -0,0 +1,245 @@
+                    getSink(),
+                    connected.remoteAddress()));
+        }
+
+        /**
+         * Constructor.
+         *
+         * @param source The {@link TcpLineSource} to send notifications through.
+         */
+        /* package private */ TcpListenerActor(final TcpLineSource source) {
+            super(source);
+        }
+    }
+
+    /**
+     * Internal actor to processRecords requests.
+     */
+    /* package private */ static final class TcpRequestHandlerActor extends UntypedActor {
+
+        /* package private */ TcpRequestHandlerActor(
+                final TcpLineSource sink,
+                final InetSocketAddress remoteAddress) {
+            _sink = sink;
+            _remoteAddress = remoteAddress;
+        }
+
+        @Override
+        public void onReceive(final Object message) throws Throwable {
+            if (message instanceof Tcp.Received) {
+                final ByteString data = ((Tcp.Received) message).data();
+
+                LOGGER.trace()
+                        .setMessage("Tcp data received")
+                        .addData("name", _sink.getName())
+                        .addData("remoteAddress", _remoteAddress.getAddress().getHostAddress())
+                        .addData("remotePort", _remoteAddress.getPort())
+                        .addData("data", data)
+                        .log();
+
+                try {
+                    processData(data);
+                    // CHECKSTYLE.OFF: IllegalCatch - Ensure all exceptions are logged (this is top level)
+                } catch (final RuntimeException e) {
+                    // CHECKSTYLE.ON: IllegalCatch
+                    BAD_REQUEST_LOGGER.warn()
+                            .setMessage("Error processing data")
+                            .addData("name", _sink.getName())
+                            .addData("remoteAddress", _remoteAddress.getAddress().getHostAddress())
+                            .addData("remotePort", _remoteAddress.getPort())
+                            .addData("data", data)
+                            .setThrowable(e)
+                            .log();
+                }
+            } else if (message instanceof Tcp.ConnectionClosed) {
+                getContext().stop(getSelf());
+                LOGGER.debug()
+                        .setMessage("Tcp connection close")
+                        .addData("name", _sink.getName())
+                        .addData("remoteAddress", _remoteAddress.getAddress().getHostAddress())
+                        .addData("remotePort", _remoteAddress.getPort())
+                        .log();
+            }
+        }
+
+        private void processData(final ByteString data) {
+            // Process datagrams line by line buffering any partial lines
+            int indexStart = 0;
+            int indexEnd = data.indexOf('\n');
+
+            // It is possible for there to be more than one line in a datagram
```

Comment:
nit: These aren't datagrams.  They're an OS buffer that represents the contents of one or more packets.

Reply:
Fixed.